### DISCUSSION TITLE
asyncio extension in SA 1.4 vs SA 2.0

### DISCUSSION THREAD

**Discussion by jeremy-patreon** [sqlalchemy/sqlalchemy#9270] [Usage Questions] [answered]
We plan to adopt asyncio extension in SA 1.4.46. According to the SA 1.4 doc,  `The asyncio extension as of SQLAlchemy 1.4.3 can now be considered to be beta level software. API details are subject to change however at this point it is unlikely for there to be significant backwards-incompatible changes.`

I wonder if asyncio extension is still considered as beta in SA 1.4.46? What's the major differences between SA 1.4 and SA 2.0 for asyncio extension?
We are considering 2 options:
1) adopt asyncio in 1.4 then migrate it to 2.0 -- if the asyncio extension is already in stable state.
2) migrate it to 2.0 first then start adopting asyncio. -- if asyncio extension in 2.0 is more stable.

**Comment by CaselIT** [comment, marked as answer]
Hi,

not much has changed on that front, just some polishing.
There haven't been particular issues reported on the main async features in the 1.4 series, so I think either strategy would be fine.
